addons/account/models/account.py

```diff
@@ -138,16 +137,26 @@ def write(self, vals):
         return rslt
 
     def unlink(self):
-        self._delete_tags_from_taxes(self.mapped('tag_ids.id'))
+        self._remove_tax_used_only_by_self()
         children = self.mapped('children_line_ids')
         if children:
             children.unlink()
         return super(AccountTaxReportLine, self).unlink()
 
+    def _remove_tax_used_only_by_self(self):
+        """ Deletes and removes from taxes and move lines all the
+        tags from the provided tax report lines that are not linked
+        to any other tax report lines.
+        """
+        all_tags = self.mapped('tag_ids.id')
+        tags_to_unlink = all_tags.filtered(lambda x: not (x.tax_report_line_ids - self))
+        self.write({'tag_ids': [(2, tag.id, 0) for tag in tags_to_unlink]})
+        self._delete_tags_from_taxes(tags_to_unlink)
```

Comment:
Shouldn't be tags written after the deletion?

Reply:
Also, in case of unlink, this is strange to write something before unlinking the records :x

Reply:
@smetl Strange, but necessary and cleaner written this way: tags can be shared between multiple report lines (in multiple reports); you don't want to delete them every time.

Also, I don't understand your first comment :p

Reply:
Forget the first comment.
Can you also add a test plz?